### DISCUSSION TITLE
Press and hold action for button

### DISCUSSION THREAD

**Discussion by riban-bw** [InfiniTimeOrg/InfiniTime#1988] [Ideas]
The button currently has two actions:

- Short press used by most apps, mostly for exit / return
- Very long press (~7s) to reset / reboot / restart watch

It would be useful to have an intermediate _long press_ action which could by default return to the main watchface but individual apps could override if they desired. It can take several presses of the button to return to the watchface / home screen and I would like to be able to get there with a single (>1.5s) press. I think this would be intuitive to many users. A _BACK_ button that has a _bold press_ action of returning home. Timing could be short<1.5<long<7<reset. This makes the operation quite manageable avoiding accidental wrong press.

**Comment by FintasticMan** [comment]
There is already a long press action, which goes directly to the watch face. So currently it's: short - back, long - watch face, longer - about, very long - reset.

**Comment by riban-bw** [comment]
Yay! I didn't see this in the docs but that is perfect, thanks! The docs do mention the long but not the longer press.
